### PR TITLE
feat: hybrid cjs and esm support

### PR DESCRIPTION
**Pull request by Nesopie**
This PR adds support for cjs and esm modules.

### CODE REVIEW

File: ts_src/index.ts [hunk 1]
```diff
@@ -26,8 +26,7 @@ function base (ALPHABET: string): base.BaseConverter {
   const iFACTOR = Math.log(256) / Math.log(BASE) // log(256) / log(BASE), rounded up
 
   function encode (source: Uint8Array | number[]): string {
-    if (source instanceof Uint8Array) {
-    } else if (ArrayBuffer.isView(source)) {
+    if (ArrayBuffer.isView(source)) {
```
Comment (junderw) [inline ts_src/index.ts:29]: I agree that this was originally poorly written, but this change makes the code different.

What if source is BOTH `Uint8Array` AND one of the other `else if`?

If you are 100% confident that these conditions will never overlap in any JS runtime ever, then maybe this is OK. But I would rather just leave it as is... or at least re-write it with the same semantic meaning... like wrapping the whole if else chain with a negation of the empty if.

Comment (Nesopie) [reply ts_src/index.ts:29]: Actually, I'm wondering if I should remove the first if statement. The function just accepts `Uint8Array | number[]` clearly.

**Comment by junderw** [comment]
npm v6 doesn't support `"lockfileVersion": 3`

File: ts_src/index.ts (hunk 1 above)
Comment (Nesopie) [reply ts_src/index.ts]: Reverted back to the original code

File: .github/workflows/main_ci.yml [hunk 2]
```diff
@@ -10,30 +10,30 @@ jobs:
   unit:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v4
         with:
-          node-version: 12
+          node-version: 17
```
Comment (junderw) [inline .github/workflows/main_ci.yml:16]: The commit says 18 but it's 17 (EOL)

**Review by junderw — APPROVED** [review]

**Pull request merged**